### PR TITLE
Add django 2.0 compatibility

### PR DESCRIPTION
**Pull request by ricard33**

### CODE REVIEW

File: PyInstaller/hooks/hook-django.py [hunk 1]
```diff
@@ -90,14 +112,16 @@
...
     # Copy migration files.
-    for mod in migration_modules:
-        mod_name, bundle_name = mod.split('.', 1)
-        mod_dir = os.path.dirname(get_module_file_attribute(mod_name))
-        bundle_dir = bundle_name.replace('.', os.sep)
-        pattern = os.path.join(mod_dir, bundle_dir, '*.py')
-        files = glob.glob(pattern)
-        for f in files:
-            datas.append((f, os.path.join(mod_name, bundle_dir)))
+    datas += collect_modules_files(migration_modules)
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-django.py]: As far as I can see, the removed lines removed are just moved into the function above. Why not just use `utils.hookcollect_submodules()` here and drop the whole function?

Anyway: Since this change is unrelated to Django 2.0, it should go into a separate commit.

File: PyInstaller/hooks/hook-django.py [hunk 2]
```diff
@@ -90,14 +112,16 @@
...
+
+    # Bundle django command scripts as data files.
+    # They are necessary get commands working.
+    logger.info('Collecting Django commands scripts.')
+    commands_modules = ['django.core.management.commands']
+    commands_modules.extend(set(app + '.management.commands' for app in installed_apps))
+    # Copy commands files.
+    datas += collect_modules_files(commands_modules)
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-django.py]: I assume, this change is solving some other [django related issues](https://github.com/pyinstaller/pyinstaller/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+django), eg.g #2313, and is unrelated to Django 2.0. So it should go into a separate commit.

File: PyInstaller/utils/hooks/subproc/django_import_finder.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #-----------------------------------------------------------------------------
-# Copyright (c) 2005-2018, PyInstaller Development Team.
+# Copyright (c) 2005-2017, PyInstaller Development Team.
```
Comment (htgoebel) [inline PyInstaller/utils/hooks/subproc/django_import_finder.py]: No time-machine :-)

File: PyInstaller/utils/hooks/subproc/django_import_finder.py [hunk 4]
```diff
@@ -100,7 +100,7 @@ def find_url_callbacks(urls_module):
     hiddenimports.append(app_ctx_proc_module)
 
 
-from django.core.urlresolvers import RegexURLPattern, RegexURLResolver
+from django.urls import URLPattern, URLResolver
```
Comment (htgoebel) [inline PyInstaller/utils/hooks/subproc/django_import_finder.py]: Please try to keep this backward-compatible. Maybe like this:
```
try:
    # django 2.x
    from django.urls import URLPattern, URLResolver
except ImportError:
    # django 1.x
    from django.core.urlresolvers import RegexURLPattern as URLPattern, \
             RegexURLResolver as URLResolver
```
In a vew year we can then remove the django 1.0 part.

File: PyInstaller/hooks/hook-django.py (hunk 1 above)
Comment (ricard33) [reply PyInstaller/hooks/hook-django.py]: This code refactoring is necessary to be able to do the same for commands_modules in lines bellow.

I'll have a look to `utils.hookcollect_submodules()` if it can do the job

**Comment by ricard33** [comment]
This MR contains all changes I've done to make it work with our software.
In fact, I think only changes in `django_import_finder.py` is needed for django 2 compatibility.

For other changes, they may be only needed by our software... I don't know. And sadly I never succeeded in making UnitTest work on my PC, so it's hard for me to prepare a clean MR...

**Comment by htgoebel** [comment]
It should be possible to provide a cleaned up pull-request even without the unittests running :-) I'm looking forward to your update.

Regarding the test suite: https://pyinstaller.readthedocs.io/en/v3.3.1/development/testing.html

**Comment by karthikdesai** [comment]
*ping* Is there any updates on this?
Planning to take this forward. If i have any updates, will report back and maybe create a new PR for it

**Comment by bjones1** [comment]
I didn't realize someone had started work on this hook, and implemented something similar that makes Django 2.0 tests pass. See https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/hooks/hook-django.py.

Would someone be willing to update this with a better approach, or provide tests that show what needs fixing? Until then, I'm closing this PR.
